docs/manual/must-call-checker.tex

```diff
@@ -0,0 +1,186 @@
+
+%% In many programs, aliasing creates two or more program elements that represent the same
+%% underlying object on which some methods might need to be called (i.e. whose type has a non-empty
+%% \<@MustCall> qualifier). For example, an \<OutputStream> might be stored in both
+%% a local variable and a field. Typically, the Must Call Checker will over-approximate both
+%% the local variable and the field (and any other pointers to the object) as \<@MustCall(\{"close"\})>:
+%% that is, because all \<OutputStream> objects might need to be closed, each pointer to an
+%% \<OutputStream> might need to be closed, too. In practice, however, some of these pointers can be
+%% ignored: they are \emph{non-owning}. In the example of an \<OutputStream> that is stored in both
+%% a local variable and a field, for instance, it may be the case that the field is non-owning (such
+%% as a cache) and the local is owning; or, it might be the case that the field is owning (an open connection,
+%% for example, that will be closed later), and the local is non-owning.
+
+%% The Must Call Checker supplies two annotations to support this concept: \<@Owning> and \<@NotOwning>.
+%% The Must Call Checker's support for these annotations is limited: method parameters annotated as
+%% \<@NotOwning> are treated as bottom (i.e. \<@MustCall(\{\})>) within the method bodies in which they
+%% appear. A client analysis of the Must Call Checker can use these annotations as a guide for which
+%% pointer to a given object is intended by the programmer as the pointer through which the must-call obligations
+%% of the object are satisfied, but these annotations are only hints.
+
+%% This feature can be disabled by passing the \<-AnoLightweightOwnership> command-line parameter to the Must
+%% Call Checker.
+
+
+\sectionAndLabel{Strings cannot have must-call obligations}{must-call-impl}
+
+This type system should not be used to track a must-call obligation for a \<String>, because it treats all
+String concatenations as having the type \<@MustCall(\{\})>.  (It does so
+even if you change the annotated JDK or a stub file to add a
+\<@MustCall(...)>\@ annotation to the declaration of \<String>.)
+
+This special-casing is because an object is sometimes implicitly converted
+to a \<String>, which inherits the object's annotations.
+Consider the following example:
+\begin{Verbatim}
+    @MustCall("close") Socket sock = ...;
+    String str = "here is sock.toString(): " + sock;
+\end{Verbatim}
+\noindent
+Without special-casing string concatenations, the type of
+\<str> would be \<@MustCall(\{"close"\})>.  This is not unsound, but it
+does lead to imprecision since the string does not need to be closed.
+
+
+\sectionAndLabel{Type parameters with implicit vs explicit bounds sometimes cause (fixable) false positives}{must-call-type-params}
+
+The defaulting rules for \<@MustCall> sometimes produce unexpected checker
```

Comment:
These warnings are unexpected to a user who has not read this section or https://checkerframework.org/manual/#climb-to-top carefully. I want this section to be searchable for a user who encounters this situation, and the symptom of this *is* an unexpected checker warning. 

You're of course correct that it's unexpected because the user misunderstands what the specification means, but I think that's beside the point (and in a section like this, which is intended to help a user who has encountered a common problem, is unnecessarily pedantic).

Reply:
I think your hypothesis is that someone who is having trouble will search the manual for the word "unexpected".  I don't share that hypothesis.  I think it's better to focus the manual's wording on facts rather than a putative mental state (which you would have to explain in the manual).